client/Components/CodeEditor/EditorConfigFunction.js

```diff
@@ -0,0 +1,35 @@
+const configEditor = function(editor, editorSession, editorTheme) {
+  editor.setTheme(`ace/theme/${editorTheme}`)
+  editorSession.setMode('ace/mode/javascript')
```

Comment:
If editorSession.setMode('ace/mode/javascript') is the way that we dictate to the editor the language that we're using, I wonder if we can create another field on state for language, and write in a <select> form where a user picks a language, and then we can do something like editorSession.setMode(`ace/mode/${this.state.language}`)

Reply:
Yea definitely with eval, we are vey limited to using Javascript, and more so using vanilla Javascript. The vm2 module I've mentioned would allow us to run non-vanilla Javaascript, i.e. Javascript with packages.

In terms of other langauges, two things:

- The child process thing Eliot mentioned to us when talking to him before class once might allow for different languages
- What I think is the ultamate solution but not sure will be feasible based on time/complexity will be spinning up a Docker environment like CircleCi does whenever the run code button is pressed